From akpm@osdl.org Tue Apr 18 21:31:11 2006
Message-Id: <200604190431.k3J4V3Bb008744@shell0.pdx.osdl.net>
Subject: Fix OCFS2 warning when DEBUG_FS is not enabled
To: greg@kroah.com
Cc: akpm@osdl.org, khali@linux-fr.org, arjan@infradead.org,
   Joel.Becker@oracle.com, mark.fasheh@oracle.com
From: akpm@osdl.org
Date: Tue, 18 Apr 2006 21:30:22 -0700


From: Jean Delvare <khali@linux-fr.org>

Fix the following warning which happens when OCFS2_FS is enabled but
DEBUG_FS isn't:

fs/ocfs2/dlmglue.c: In function `ocfs2_dlm_init_debug':
fs/ocfs2/dlmglue.c:2036: warning: passing arg 5 of `debugfs_create_file' discards qualifiers from pointer target type

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Cc: Arjan van de Ven <arjan@infradead.org>
Cc: Joel Becker <Joel.Becker@oracle.com>
Acked-by: Mark Fasheh <mark.fasheh@oracle.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

---
 include/linux/debugfs.h |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/include/linux/debugfs.h
+++ gregkh-2.6/include/linux/debugfs.h
@@ -58,9 +58,8 @@ struct dentry *debugfs_create_blob(const
  */
 
 static inline struct dentry *debugfs_create_file(const char *name, mode_t mode,
-						 struct dentry *parent,
-						 void *data,
-						 struct file_operations *fops)
+					struct dentry *parent, void *data,
+					const struct file_operations *fops)
 {
 	return ERR_PTR(-ENODEV);
 }
